src/groupeddataframe/grouping.jl

```diff
@@ -1002,10 +1007,10 @@ function groupreduce!(res, f, op, condf, adjust,
     end
     outcol = adjust === nothing ? res : map(adjust, res, counts)
     # Undo pool sharing done by groupreduce_init
-    if outcol isa CategoricalVector
+    if outcol isa CategoricalVector && outcol.pool === incol.pool
```

Comment:
I do not see where this "unnecessary" copy is made. It seems that the check you do here is needed. Also it should not be very expensive - right?

Reply:
This change is actually just a consequence of the fact that a copy is made: the pool can be in a different order after `map`. The unnecessary copy is due to passing `adjust !== nothing`, which induces a call to `map`.

Reply:
OK. The changes also look good.